### PR TITLE
Pipenet fix

### LINKED ISSUE #18354: four-way manifolds are broken
they don't fill with gas when connected to any type of pump; they are a brick wall to any atmos setup which means that anything past the four-way manifold will not fill no matter what is attached to it.

regular manifolds work fine.

### LINKED ISSUE #36298: Pipenet bugs
Server revision compiled on: 2018-03-08
8047b0
Revision: c07abf4f9eca6df3430f6bb72c942abaacb9ae31
1. Build this.
scrnshot1
2. Add pipe
scrnshot2
3. Wrench canister to it and see unvisible disconnection. (result of "Runtime in components_base.dm,91")
scrnshot3
4. Here i replace the connection port and fill pipe by N2O.
scrnshot4
5. Unwrench pipe and see no n2o in air.
scrnshot5
6. Wrench pipe back and see empty pipe.
scrnshot6

### PR DESCRIPTION
**Pull request by Dennok**
#Fix of Runtime in components_base.dm,91: Cannot read null.parents  proc name: returnPipenet
Return working releaseAirToTurf() on pipe Destroy()
Block of self merge what cause pipeline annihilation. Pipes don't lose air on connecting more than to one pipe

:cl: Pipe fixes
fix: Runtime in components_base.dm,91 what cause atmos machinery connection problems.
fix: Pipes release air in turf on Destroy(), again.
fix: Pipeline now cant be annihilated by merge(src). Pipes don't lose air on connecting more than to one pipe.
/:cl:

[why]: # (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)

Fixes #36298
Fixes #18354

### CODE REVIEW

File: code/modules/atmospherics/machinery/components/components_base.dm [hunk 1]
```diff
@@ -88,6 +88,8 @@ Pipenet stuff; housekeeping
 	parents[nodes.Find(A)] = reference
 
 /obj/machinery/atmospherics/components/returnPipenet(obj/machinery/atmospherics/A = nodes[1]) //returns parents[1] if called without argument
+	if(QDELETED(src))
+		warning(" returnPipenet called on QDELETED [src]  Nearby: ([x], [y], [z])")
 	return parents[nodes.Find(A)]
```
Comment (Dennok) [inline code/modules/atmospherics/machinery/components/components_base.dm]: Don't know why it fix "Runtime in components_base.dm,91", but it fix it.

**Comment by ExcessiveUseOfCobblestone** [comment]
can this get a bypass on the 24 hour? I know it can absolutely ruin the SM.

File: code/modules/atmospherics/machinery/datum_pipeline.dm [hunk 2]
```diff
@@ -90,7 +90,8 @@
 		var/obj/machinery/atmospherics/pipe/P = A
 		if(P.parent)
 			merge(P.parent)
-		P.parent = src
+		else
```
Comment (duncathan) [inline code/modules/atmospherics/machinery/datum_pipeline.dm]: why did you add this `else` statement?

File: code/modules/atmospherics/machinery/components/components_base.dm (hunk 1 above)
Comment (duncathan) [reply code/modules/atmospherics/machinery/components/components_base.dm]: I strongly suspect that this is not fixing the bug, just hiding the runtime. Can you demonstrate otherwise?

**Review by duncathan — COMMENTED** [review]
the changes I didn't comment on are good, ty for your contribution

**Comment by duncathan** [comment]
@ExceaaiveUseOfCobblestone not a chance

**Comment by ExcessiveUseOfCobblestone** [comment]
luckily you didn't review until >24 hours anyways :^)

File: code/modules/atmospherics/machinery/datum_pipeline.dm (hunk 2 above)
Comment (Dennok) [reply code/modules/atmospherics/machinery/datum_pipeline.dm]: `<else>` is not necessary here, just removed one extra assignation.

**Comment by Dennok** [comment]
Fix of 36298

**Comment by Dennok** [comment]
Fix of https://github.com/tgstation/tgstation/issues/18354

File: code/modules/atmospherics/machinery/components/components_base.dm (hunk 1 above)
Comment (Dennok) [reply code/modules/atmospherics/machinery/components/components_base.dm]: Here is a chain of 2 runtimes, one launch another herehttps://github.com/tgstation/tgstation/pull/36230/files/aa5dca9a52f10125d9214227f700fb5036d72700#diff-8880ff6809e0203d86f05941e08972cfL130
now it dont launches.
https://github.com/tgstation/tgstation/issues/36298
i test all (511) and 3 point work propertly.

File: code/modules/atmospherics/machinery/datum_pipeline.dm (hunk 2 above)
Comment (duncathan) [reply code/modules/atmospherics/machinery/datum_pipeline.dm]: revert this change; that assignment is not redundant

Comment (Dennok) [reply code/modules/atmospherics/machinery/datum_pipeline.dm]: merge already have '<P.parent = src>'

**Comment by ShizCalev** [comment]
@duncathan Opinion?

**Comment by vuonojenmustaturska** [comment]
i have backed out the conditional `warning()` that cannot actually have fixed the runtime, at best it is hiding it due to a byond logging bug
